lib/CachedResolver.js

```diff
@@ -20,15 +22,20 @@ export class CachedResolver {
     this._cache = new LruCache({max, maxAge, updateAgeOnGet, ...cacheOptions});
     this._methods = new Map();
   }
-
-  use(driver) {
-    const methodName = driver.method;
+  /**
+   * Registers a driver for a did:method.
+   *
+   * @param {object} driver - A did method driver.
+   * @param {string} [method] - An optional method.
+   */
+  use(driver, method) {
```

Comment:
We've never used that pattern for registering plugins or middleware (I forget what the name of this pattern is) so I just stuck with what we already have. I might add that method parameter exists for one reason: test data this might be over thinking an optional parameter.